source/aggregation-pipeline-builder.txt

```diff
@@ -73,8 +73,62 @@ Limitations
 Create an Aggregation Pipeline
 ------------------------------
 
+Aggregation pipelines can be created in 2 seperate modes:
+
+- Stage View Mode, a visual click based pipeline editor which 
+  preloads pipeline syntax based on user selections.
+
+- Text View Mode, a text based pipeline editor which accepts raw
+  pipeline syntax. 
+
+.. _create-agg-pipeline-stageview:
+
+Stage View Mode
+~~~~~~~~~~~~~~~
+|
+
 .. include:: /includes/steps/create-agg-pipeline.rst
 
+.. _create-agg-pipeline-textview:
+
+Text View Mode
+~~~~~~~~~~~~~~
+|
+
+.. procedure::
+   :style: normal
+
+   .. step:: In the aggregation pipeline pane, click the 
+      :guilabel:`</> Text` toggle switch to enable text mode for 
+      pipeline editing.
+
+      .. figure:: /images/compass/agg-builder-textview-toggle-on.png
+         :alt: Toggle textmode on
+
+   .. step:: Enter valid aggregation syntax in the newly prompted 
+      text editor. The following example pipeline limits the query 
+      results to ``4`` documents.
```

Comment:
[minimalism]

I think we can omit "newly prompted" here. Removing it might make this read a bit more smoothly.